AudiusProject/audius-client
# [C-2348] Change offline download location

### Description

Change offline download location from Audius Documents directory to the Cache directory. This hides the files from the iOS file browser.

Also adds a best-effort sync process that will copy files from the old to new locations before loading the offline data from disk. On failure, the sync will abort, delete the old data, and the sync process will resume downloads in the new location.

### Dragons

Putting the wrong start/end directories can wipe users's existing downloads on first load.

### How Has This Been Tested?

iOS simulator & android simulator
1. Download tracks on old build
2. Switch to new build
3. Check for migration debug logs
4. Assert tracks are no longer visible in file browser and are visible in app offline
